pages/blog/websocket-part2.md

```diff
@@ -0,0 +1,833 @@
+                status: online
+                version: 1.0.0
+```
+
+### Describe responses - specification extensions
+
+Describe responses? What responses? 
+
+It is EDA. Who cares about responses, right? Fire and forget rules!
+
+The thing is that request and reply pattern is also used in EDA. This is also the case with Kraken API where communication goes through a single channel with multiple different messages. One message triggers another message in response.
+
+The simplest example is the message **ping** that triggers a **pong** reply. The current AsyncAPI limitation is that you cannot specify that once the user sends (publish) message **ping**, the **pong** message is received in a reply. Look at this [thread](https://github.com/asyncapi/spec/issues/94) to participate in an ongoing discussion about request/reply pattern support in AsyncAPI.
+
+For **docs-only** road from section [Choosing the right road to Rome](#choosing-the-right-road-to-rome), I would be lazy and just put such info in the description of both messages. Even though this is an error-prone approach, I would just make my life easier. For **automation** road I would choose to use a specification extension.
+
+What is specification extension?
+
+You can extend every AsyncAPI object in the AsyncAPI document with extra properties. You only need to prefix them with `x-`. You can also share extensions or reuse extensions from others thanks to [extensions catalog](https://github.com/asyncapi/extensions-catalog).
+
+In the below document, you will notice that for the request/reply pattern, I use AsyncAPI specification extensions called **x-response**.
+
+```yml
+messages:
+    ping:
+        summary: Ping server to determine whether connection is alive
+        description: Client can ping server to determine whether connection is alive, server responds with pong. This is an application level ping as opposed to default ping in websockets standard which is server initiated
+        payload:
+            $ref: '#/components/schemas/ping'
+        x-response:
+            $ref: '#/components/messages/pong'
+```
+
+Even though the reference to another object is provided inside the extension that is not part of AsyncAPI, our parser will resolve it correctly. It means that under **x-response** property, I will have access to the entire message object.
+
+### Schemas vs JSON Schema
+
+Because the message itself is most important in the entire EDA, you need to describe the message payload properly. 
+
+AsyncAPI allows you to provide payload information in different schema formats. The default format is AsyncAPI Schema that is a superset of JSON Schema. You can use others too, like Avro, for example.
+
+From the AsyncAPI document point of view, the most important is that you can reuse schemas. In other words, instead of providing data directly to the **payload** object, you can **$ref** them from **components.schema** or even an external document. Just DRY, right?
```

Comment:
```suggestion
From the AsyncAPI document point of view, the most important is that you can reuse schemas. In other words, instead of providing data directly to the **payload** object, you can **$ref** them from **components.schemas** or even an external document. Just DRY, right?
```